app/controllers/contributions_controller.rb

```diff
@@ -42,13 +43,37 @@ def show
       format.rdf { render xml: contribution.to_rdfxml }
       format.ttl { render plain: contribution.to_turtle }
     end
+  rescue Mongoid::Errors::DocumentNotFound
+    Contribution.deleted.find_by(oai_pmh_record_id: params[:uuid])
+    render plain: Rack::Utils::HTTP_STATUS_CODES[410], status: 410
   end
 
   def edit
-    cho = EDM::ProvidedCHO.find_by(uuid: params[:uuid])
-    contribution = cho.edm_aggregatedCHO_for.contribution
+    contribution = contribution_from_params
     authorize! :edit, contribution
-    redirect_to send(:"edit_#{contribution.campaign.dc_identifier}_path", cho)
+    redirect_to send(:"edit_#{contribution.campaign.dc_identifier}_path", params[:uuid])
+  end
+
+  def delete
+    @contribution = contribution_from_params
+    authorize! :wipe, @contribution
+  end
+
+  def destroy
+    contribution = contribution_from_params
+    authorize! :wipe, contribution
+    begin
+      if contribution.ever_published?
+        contribution.wipe!
+        flash[:notice] = I18n.t('contribute.contributions.notices.wiped', name: contribution.dc_title.join('; '))
```

Comment:
Let's avoid having `contribution.dc_title.join('; ')` three times and either move to a method or set in a variable before this block.